pkg/controllers/scheduler/framework/plugins/rsp/rsp.go

```diff
@@ -127,6 +112,34 @@ func (pl *ClusterCapacityWeight) ReplicaScheduling(
 		currentReplicas[cluster] = totalReplicas
 	}
 
+	clusterPreferences := map[string]planner.ClusterPreferences{}
+	for _, cluster := range clusters {
+		pref := planner.ClusterPreferences{
+			Weight:      schedulingWeights[cluster.Name],
+			MinReplicas: su.MinReplicas[cluster.Name],
+			MaxReplicas: nil,
+		}
+
+		if maxReplicas, exists := su.MaxReplicas[cluster.Name]; exists {
+			pref.MaxReplicas = pointer.Int64(maxReplicas)
+		}
+
+		// if member cluster has untolerated NoSchedule taint, no new replicas will be scheduled to this cluster
+		if _, isUntolerated := framework.FindMatchingUntoleratedTaint(
+			cluster.Spec.Taints,
+			su.Tolerations,
+			func(t *corev1.Taint) bool {
+				return t.Effect == corev1.TaintEffectNoSchedule
+			},
+		); isUntolerated {
+			if pref.MaxReplicas == nil || (pref.MaxReplicas != nil && currentReplicas[cluster.Name] < *pref.MaxReplicas) {
```

Comment:
the `||` will be shortcircuited if `pref.MaxReplicas` == nil, so the nil check is redundant.
```suggestion
			if pref.MaxReplicas == nil || currentReplicas[cluster.Name] < *pref.MaxReplicas {
```

Reply:
The code has been updated.